compiler/damlc/daml-doc/src/DA/Daml/Doc/HaddockParse.hs

```diff
@@ -136,15 +148,33 @@ data DeclData = DeclData
 
 buildDocCtx :: TypecheckedModule -> DocCtx
 buildDocCtx dc_tcmod  =
-  let dc_mod
-          = Modulename . T.pack . moduleNameString . moduleName
-          . ms_mod . pm_mod_summary . tm_parsed_module $ dc_tcmod
+  let dc_mod = packModule . ms_mod . pm_mod_summary . tm_parsed_module $ dc_tcmod
       dc_decls
           = map (uncurry DeclData) . collectDocs . hsmodDecls . unLoc
           . pm_parsed_source . tm_parsed_module $ dc_tcmod
       (dc_templates, dc_choices)
           = getTemplateData . tm_parsed_module $ dc_tcmod
 
+      tythings = modInfoTyThings . tm_checked_module_info $ dc_tcmod
+
+      dc_tycons = MS.fromList . catMaybes . flip map tythings $ \case
+          ATyCon tycon ->
+              let typename = Typename . packName . tyConName $ tycon
+              in Just (typename, tycon)
+          _ -> Nothing
+
+      dc_datacons = MS.fromList . catMaybes . flip map tythings $ \case
+          AConLike (RealDataCon datacon) ->
+              let conname = Typename . packName . dataConName $ datacon
+              in Just (conname, datacon)
+          _ -> Nothing
+
+      dc_ids = MS.fromList . catMaybes . flip map tythings $ \case
+          AnId id ->
```

Comment:
Depending on how many `tythings` we have, this could be sped up by only doing a single traversal but at this point I highly doubt it’s worth it.

Reply:
Single pass seems more trouble than its worth, tbh.